src/app/domain/offline/hooks/useOfflineReplicationOnRealtime.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect } from 'react'
+
+import { useClient } from 'cozy-client'
+import type { CozyClientDocument } from 'cozy-client/types/types'
+import Minilog from 'cozy-minilog'
+
+import { triggerPouchReplication } from '/app/domain/offline/utils'
+import { offlineDoctypes } from '/pouchdb/getLinks'
+
+const log = Minilog('📶 useOfflineReplicationOnRealtime')
+
+export const useOfflineReplicationOnRealtime = (): void => {
+  const client = useClient()
+
+  useEffect(() => {
+    if (!client) return
+
+    // @ts-expect-error client.plugins is not typed
+    const realtime = client.plugins.realtime as CozyRealtime
+
+    const triggerReplication =
+      (verb: string) =>
+      (doc: CozyClientDocument): void => {
+        const docInfo = `${verb} ${doc._type ?? ''} ${doc._id ?? ''}`
+        log.debug(`Trigger replication from realtime event (${docInfo})`)
+        triggerPouchReplication(client)
+      }
+
+    const triggerReplicationCreated = triggerReplication('created')
+    const triggerReplicationUpdated = triggerReplication('updated')
+    const triggerReplicationDeleted = triggerReplication('deleted')
+
+    offlineDoctypes.forEach(doctype => {
+      realtime.subscribe('created', doctype, triggerReplicationCreated)
+      realtime.subscribe('updated', doctype, triggerReplicationUpdated)
+      realtime.subscribe('deleted', doctype, triggerReplicationDeleted)
```

Comment:
I realize we trigger a full replication on all doctypes for any realtime change, while we probably should only replicate for the concerned doctype, to save resources. But this is an improvement that can be made for later

Reply:
Yes, we will have to implement something to handle that scenario. For now the replication method always replicate all pouches and no mechanism has been implemented to debounce replications separately.